apps/web/src/server/app/validation/validation.controller.js

```diff
@@ -354,6 +354,59 @@ export function getCheckAndConfirm(request, response) {
 	});
 }
 
+/**
+ * Utility function to create Reason data in valid format for updateAppeal API post using the supplied details data
+ *
+ * @param {string} outcomeType - string specifying the applicable outcome ('invalid' or 'incomplete') (valid outcome returns an empty object as Reasons not required for that case)
+ * @param {object} appealDetails - details object from session storage containing data on the outcome
+ * @returns {object} - resulting Reasons object in valid format (see /api-docs/#/default/post_validation__id_ for details)
+ */
+function formatOutcomeReasonsDataForApiPost(outcomeType, appealDetails) {
+	let data = {};
+
+	switch (outcomeType) {
+		case 'invalid':
+			data = {
+				// eslint-disable-next-line unicorn/no-array-reduce, unicorn/prefer-object-from-entries
+				...appealDetails.invalidReasons.reduce((previous, current) => {
```

Comment:
Minor thing as there's no technical impact, but this naming is a bit confusing as it doesn't really match the signature of reduce. By its name, "current" would suggest a new representation of "previous", whereas it is in fact a key for the reason, while "previous" is the map being reduced.

Reply:
I agree these could have been clearer (this is the terminology used in the reducer examples on MDN which is why I renamed the original single-letter variables to these). I've updated to "accumulator" and "value" now.